### PR TITLE
Jv update cow health job

### LINKED ISSUE #59: Adjust cow health over time (in the Cow Health Update Job)
- Every time the Cow Health Job runs, the commons' average cow health changes depending on whether a cow threshold is passed.

You will need to iterate over each of the commons, and each of the users, and update the cow health accordingly.

The number of cows here is the total number in each commons.

Here is the formula for adjusting cow health:

1.  Define int carryingCapacity = 100, unless the PR that adds carrying capacity to the commons has already been merged. In that case, let carryingCapacity be commons.getCarryingCapacity();
2.  If commons.getTotalCows() <= carryingCapacity, then increase average cow health (expressed as a percentage) by threshold, but do not let it exceed 100%.
3. Else (i.e. commons.getTotalCow() > carryingCapacity, then reduce average cow health (expressed as a percentage) by min( (cow - carryingCapacity) * threshold , 100 ). That is, whatever the threshold value is, that value is the percentage by which the averageCowHealth goes down, for each cow over the carrying capacity.

Acceptance Criteria

- [ ] Every time the cow health job is run, average cow health for each user in the commons goes up or down by the amount specified by the formula.

### PR DESCRIPTION
**Pull request by heyyysus**
In this PR we update the "updateCowHealthJob" in the backend of the application to do the following for each commons:
- If the total number of cows in the commons is under the carrying capacity, increase the health of each userCommons average linked to that commons by the degradationRate. 
- Else, decrease the health of each userCommons average linked to that commons by the degradationRate * (totalNumCows - carryingCapacity)

We also update tests. 

Closes #59

### CODE REVIEW

**Review by sergioacolis — DISMISSED** [review]
Tested on Heroku using swagger, cows lose health when over the carrying capacity, and gain health when under the carrying capacity as expected, LGTM

File: src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java [hunk 1]
```diff
@@ -1,33 +1,355 @@
...
-                .build();
+
+        UpdateCowHealthJob updateCowHealthJob = new UpdateCowHealthJob(commonsRepository, userCommonsRepository);
         updateCowHealthJob.accept(ctx);
 
         // Assert
 
         String expected = "Starting to update cow health.\n" +
-                "This is where the code to update the cow's health will go.\n" +
+               // "This is where the code to update the cow's health will go.\n" +
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java:63]: We don't leave commented out code in PRs that are intended to be merged into main.

Please don't do this when you end up at an internship or job.  

And to whomever code reviewed this: you should have caught that.

**Review by pconrad — CHANGES_REQUESTED** [review]

File: src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java (hunk 1 above)
Comment (heyyysus) [reply src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java]: Commented out code has been removed.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**